django_keycloak/middleware.py

```diff
@@ -42,113 +82,61 @@ def append_user_info_to_request(self, request, token):
-        auth_header = request.META.get("HTTP_AUTHORIZATION").split()
-        if len(auth_header) == 2:
-            return auth_header[1]
-        return None
-
-
-class KeycloakGrapheneMiddleware(KeycloakMiddlewareMixin):
-    """
-    Middleware to validate Keycloak access based on Graphql validations
-    """
-
-    def __init__(self):
-        logging.warning(
-            "All functionality is provided by KeycloakMiddleware", DeprecationWarning, 2
-        )
-        self.keycloak = Connect()
-
-    def resolve(self, next, root, info, **kwargs):
-        """
-        Graphene Middleware to validate keycloak access
-        """
-        request = info.context
-
-        if self.is_auth_header_missing(request):
-            """Append anonymous user and continue"""
-            return next(root, info, **kwargs)
-
-        token = self.get_token(request)
-        if token is None:
-            raise Exception("Invalid token structure. Must be 'Bearer <token>'")
-
-        if not self.keycloak.is_token_active(token):
-            raise Exception("Invalid or expired token.")
-
-        info.context = self.append_user_info_to_request(request, token)
+        return "HTTP_AUTHORIZATION" in request.META
 
-        return next(root, info, **kwargs)
-
-
-class KeycloakMiddleware(KeycloakMiddlewareMixin, MiddlewareMixin):
-    """
-    Middleware to validate Keycloak access based on REST validations
-    """
-
-    sync_capable = True
-    async_capable = False
-
-    def __init__(self, get_response):
-        self.keycloak = Connect()
-
-        # Django response
-        self.get_response = get_response
-
-    def __call__(self, request):
+    def process_request(self, request):
         """
-        To be executed before the view each request
+        To be executed before the view each request.
         """
-        # Skip auth for gql endpoint (it is done in KeycloakGrapheneMiddleware)
-        if self.is_graphql_endpoint(request):
-            return self.get_response(request)
-
-        if not self.is_auth_header_missing(request):
-            token = self.get_token(request)
-            if token is None:
-                return JsonResponse(
-                    {"detail": "Invalid token structure. Must be 'Bearer <token>'"},
-                    status=401,
-                )
-
-            if not self.keycloak.is_token_active(token):
-                return JsonResponse(
-                    {"detail": "Invalid or expired token."},
-                    status=401,
-                )
-            request = self.append_user_info_to_request(request, token)
-        return self.get_response(request)
+        # Skip auth in the following cases:
+        # 1. It is a graphql endpoint (handled by KeycloakGrapheneMiddleware)
+        # 2. It is a URL in "EXEMPT_URIS"
+        # 3. Request does not contain authorization header
+        # Also skip auth for "EXEMPT_URIS" defined in configs
+        if (
+            self.is_graphql_endpoint(request)
```

Comment:
If we're removing special GraphQL treatment, this check can be removed

Reply:
Ok, consider it done